.github/workflows/ci.yml

```diff
@@ -461,6 +461,38 @@ jobs:
           cargo install --locked kani-verifier       &> /dev/null || true
           cargo kani setup                           &> /dev/null || true
 
+  check-job-dependencies:
+    runs-on: ubuntu-latest
+    name: Check all-jobs-succeeded depends on all jobs
+    steps:
+      - name: Install yq (for YAML parsing)
+        run: go install github.com/mikefarah/yq/v4@latest
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: Run dependency check
+        run: |
+          set -eo pipefail
+          jobs=$(for i in $(find .github -iname '*.yaml' -or -iname '*.yml')
+            do
+              # Select jobs that are triggered by pull request.
+              if yq -e '.on | has("pull_request")' "$i" 2>/dev/null >/dev/null
+              then
+                # This gets the list of jobs that are not in
+                # all-jobs-succeed.needs .
+                comm -23 \
+                  <(yq -r '.jobs | keys | .[]' "$i" | sort | uniq) \
+                  <(yq -r '.jobs.all-jobs-succeed.needs[]' "$i" | sort | uniq)
+              fi
+            done | sort | uniq
+          )
+          # Effectively, this checks that all-jobs-succeed does not depend on
+          # itself. Checking for the existence of a job that we don't want
```

Comment:
IIUC, it's slightly different than this - it checks that the set of jobs that all-jobs-succeeded does not depend on is _exactly_ itself. That distinction is important because it also ensures that there are no jobs _besides_ itself that it doesn't depend on. Do I understand that correctly? If so, could you update the comment?